pulpcore/cli/rpm/repository.py

```diff
@@ -165,16 +166,49 @@ def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> Non
 @name_option
 @href_option
 @remote_option
-@click.option("--mirror/--no-mirror", default=None)
 @click.option(
-    "--skip-type", "skip_types", multiple=True, type=click.Choice(SKIP_TYPES, case_sensitive=False)
+    "--mirror/--no-mirror",
+    default=None,
+    help="""
+    DEPRECATED: If True, sync_policy will default to 'mirror_complete' instead
```

Comment:
I don't know. Is "deprecated" the proper term for the cli? Should we explain, that this is only for certain old versions of the plugin? How about `needs_plugin` with an upper bound?

Reply:
It's deprecated with the same wording in the pulp_rpm REST API : https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/serializers/repository.py#L298

It's been deprecated since pulp_rpm/3.16.0. I don't want to put a max=3.16 for RPM on it, because people are using it now, and would suddenly break. We could put a max=3.17 on it, which would make it stop working when pulp_rpm/3.18 is released Really Soon Now (and I don't know if pulp_rpm is going to be able to actually remove the option any time soon)

Reply:
OK, as long as the option is there, we should support it. :+1: